src/Features/Core/Portable/CodeLens/CodeLensFindReferenceProgress.cs

```diff
@@ -51,60 +51,48 @@ internal sealed class CodeLensFindReferencesProgress : IFindReferencesProgress,
-                         && candidateSyntaxNode.SyntaxTree.Equals(sourceLocation.SourceTree)
-                         && candidateSyntaxNode.Span.Contains(sourceLocation.SourceSpan)
-                   select sourceLocation;
         }
 
-        /// <summary>
-        /// Exclude the following kind of symbols:
-        ///  1. Implicitly declared symbols (such as implicit fields backing properties)
-        ///  2. Symbols that can't be referenced by name (such as property getters and setters).
-        ///  3. Metadata only symbols, i.e. symbols with no location in source.
-        /// </summary>
-        private static bool FilterReference(ISymbol queriedSymbol, ISymbol definition, ReferenceLocation reference)
+        public void OnFindInDocumentCompleted(Document document)
         {
-            var isImplicitlyDeclared = definition.IsImplicitlyDeclared || definition.IsAccessor();
-            // FindRefs treats a constructor invocation as a reference to the constructor symbol and to the named type symbol that defines it.
-            // While we need to count the cascaded symbol definition from the named type to its constructor, we should not double count the
-            // reference location for the invocation while computing references count for the named type symbol. 
-            var isImplicitReference = queriedSymbol.Kind == SymbolKind.NamedType &&
-                                      (definition as IMethodSymbol)?.MethodKind == MethodKind.Constructor;
-            return isImplicitlyDeclared ||
-                   isImplicitReference ||
-                   (!definition.Locations.Any(loc => loc.IsInSource) && !reference.Location.IsInSource);
         }
 
         private static bool FilterDefinition(ISymbol definition)
         {
             return definition.IsImplicitlyDeclared ||
-                   (definition as IMethodSymbol)?.AssociatedSymbol != null ||
-                   !definition.Locations.Any(loc => loc.IsInSource);
+                   (definition as IMethodSymbol)?.AssociatedSymbol != null;
+        }
+
+        /// <summary>
+        /// Returns partial symbol locations whose node does not match the queried syntaxNode
+        /// </summary>
+        /// <param name="symbol">Symbol whose locations are queried</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>Partial locations</returns>
+        private IEnumerable<Location> GetPartialLocations(ISymbol symbol, CancellationToken cancellationToken)
+        {
+            // Returns nodes from source not equal to actual location
+            return from syntaxReference in symbol.DeclaringSyntaxReferences
+                let candidateSyntaxNode = syntaxReference.GetSyntax(cancellationToken)
+                where !(_queriedNode.Span == candidateSyntaxNode.Span &&
+                        _queriedNode.SyntaxTree.FilePath.Equals(candidateSyntaxNode.SyntaxTree.FilePath,
```

Comment:
IIRC - the main issue here was that people didn't want to see the definition the codelens indicator was directly above included in the count.

Tagging @DustinCampbell and @balajikris in case they have more context.